src/explore-education-statistics-frontend/src/modules/subscriptions/ConfirmSubscriptionPage.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState } from 'react';
+import { GetServerSideProps, NextPage } from 'next';
+import Button from '@common/components/Button';
+import publicationService, {
+  PublicationTitle,
+} from '@common/services/publicationService';
+import { Dictionary } from '@common/types';
+import Page from '@frontend/components/Page';
+import notificationService, {
+  Subscription,
+} from '@frontend/services/notificationService';
+import withAxiosHandler from '@frontend/middleware/ssr/withAxiosHandler';
+import SubscriptionStatusMessage from '@frontend/modules/subscriptions/components/SubscriptionStatusMessage';
+import VerificationErrorMessage from '@frontend/modules/subscriptions/components/VerificationErrorMessage';
+import Head from 'next/head';
+
+interface Props {
+  publicationSlug: string;
+  data: PublicationTitle;
```

Comment:
I'd be inclined to rename this from `data` to `publication`

Same with ConfirmUnsubscriptionPage

Reply:
Completely agreed - it also irks me that we've got the slug in two places, but I appreciate there's actually a technical benefit in that case. I only left it as-was at first because I was trying not to change too much existing code, but at this point I mean... why not :D